include/eve/module/core/regular/impl/simd/x86/is_less.hpp

```diff
@@ -19,58 +19,184 @@
+        else if constexpr (c == category::uint8x64)   return mask64 {_mm512_cmplt_epu8_mask(a, b)};
+        else if constexpr (c == category::uint8x32)   return mask32 {_mm256_cmplt_epu8_mask(a, b)};
+        else if constexpr (c == category::uint8x16)   return mask16 {_mm_cmplt_epu8_mask(a, b)};
+        else if constexpr (c == category::int64x8)    return mask8 {_mm512_cmplt_epi64_mask(a, b)};
+        else if constexpr (c == category::int64x4)    return mask8 {_mm256_cmplt_epi64_mask(a, b)};
+        else if constexpr (c == category::int64x2)    return mask8 {_mm_cmplt_epi64_mask(a, b)};
+        else if constexpr (c == category::int32x16)   return mask16 {_mm512_cmplt_epi32_mask(a, b)};
+        else if constexpr (c == category::int32x8)    return mask8 {_mm256_cmplt_epi32_mask(a, b)};
+        else if constexpr (c == category::int32x4)    return mask8 {_mm_cmplt_epi32_mask(a, b)};
+        else if constexpr (c == category::int16x32)   return mask32 {_mm512_cmplt_epi16_mask(a, b)};
+        else if constexpr (c == category::int16x16)   return mask16 {_mm256_cmplt_epi16_mask(a, b)};
+        else if constexpr (c == category::int16x8)    return mask8 {_mm_cmplt_epi16_mask(a, b)};
+        else if constexpr (c == category::int8x64)    return mask64 {_mm512_cmplt_epi8_mask(a, b)};
+        else if constexpr (c == category::int8x32)    return mask32 {_mm256_cmplt_epi8_mask(a, b)};
+        else if constexpr (c == category::int8x16)    return mask16 {_mm_cmplt_epi8_mask(a, b)};
+      }
+      else
+      {
+        if      constexpr (c == category::float32x8) return _mm256_cmp_ps(a, b, f);
+        else if constexpr (c == category::float64x4) return _mm256_cmp_pd(a, b, f);
+        else if constexpr (c == category::float32x4) return _mm_cmplt_ps(a, b);
+        else if constexpr (c == category::float64x2) return _mm_cmplt_pd(a, b);
+        else
+        {
+          constexpr auto use_avx2   = current_api >= avx2;
+          constexpr auto use_sse4_1 = current_api >= sse4_1;
+          constexpr auto lt         = []<typename E>(E ev, E fv) { return as_logical_t<E>(ev < fv); };
+
+          [[maybe_unused]] auto unsigned_cmp = [](auto lhs, auto rhs)
+          {
+            using l_t     = logical<wide<T, N>>;
+            auto const sm = signmask(as<as_integer_t<wide<T, N>, signed>>{});
+            return bit_cast((bit_cast(lhs, as(sm)) - sm) < (bit_cast(rhs, as(sm)) - sm), as<l_t>{});
+          };
+
+          if      constexpr (use_avx2 && c == category::int64x4)   return _mm256_cmpgt_epi64(b, a);
+          else if constexpr (use_avx2 && c == category::uint64x4)  return unsigned_cmp(a, b);
+          else if constexpr (use_avx2 && c == category::int32x8)   return _mm256_cmpgt_epi32(b, a);
+          else if constexpr (use_avx2 && c == category::uint32x8)
+          {
+            return _mm256_xor_si256(_mm256_cmpeq_epi32(_mm256_min_epu32(a, b), b), _mm256_cmpeq_epi32(b, b));
```

Comment:
No - just equal max. I'm not 100 what this is. Comparing self to self for equality.

Reply:
I see - sucks

> The comparison with self is there to generate a vector of 1s

You don't need to teach the compilers how to generate that, don't be clever.
Also you have andnot - you don't need this.
Also also - please write this as `a != max(a, b)` - srsly. 
Here - compiler will do everything: https://godbolt.org/z/hMbzeo1MW

Reply:
I don't think that codegen is a problem. I'd trust the compiler. There are ~no algorithms that can survive without having a zero somewhere.